keps/sig-apps/4026-job-creation-timestamp-annotation/README.md

```diff
@@ -0,0 +1,304 @@
+
+The feature will be tested manually prior to beta launch.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+N/A
+
+###### How can an operator determine if the feature is in use by workloads?
+
+- Check if CronJobs have the annotation `batch.kubernetes.io/job-creation-timestamp`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason:
+- [X] API .status
+  - Condition name:
+  - Other field:
+    - `.metadata.annotations['batch.kubernetes.io/job-creation-timestamp']`
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- 99% percentile over day for Job syncs is <= 15s for a client-side 50 QPS limit.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+New job annotation of size 34B plus value of size N where N is the number of digits in the job ordinal.
+Worst case for N would be the max number of jobs per node * max number of nodes.
+Per the docs on [large clusters](https://kubernetes.io/docs/setup/best-practices/cluster-large/), this would be 110 pods/node * 5000 nodes = 550,000 (6 digits).
+
+Hence, max annotation size would be 34 + 6 = 40B.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+N/A
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+N/A
+
+###### What are other known failure modes?
+
+N/A
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+- 2023-05-22: KEP published
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
```

Comment:
A few questions I have (mostly as a budding k8 dev):

1) Why not add this as a label instead?

2) Why not add this as a status field?  Job has a StartTime status field

Reply:
Added to the alternatives.

Reply:
Okay!  Thank you for clearing that up.